src/api/serializers.py

```diff
@@ -0,0 +1,72 @@
+from rest_framework.serializers import ModelSerializer
+
+from account.models import Customer
+from music.models import Album, Artist, Genre, Playlist, Track
+
+
+class CustomerSerializer(ModelSerializer):
+    class Meta:
+        model = Customer
+        fields = ("first_name", "last_name", "email", "password", "is_staff")
+
+
+class ArtistSerializer(ModelSerializer):
+    class Meta:
+        model = Artist
+        fields = ("id", "name")
+
+
+class AlbumSerializer(ModelSerializer):
+    class Meta:
+        model = Album
+        fields = ("id", "title")
+
+
+class TrackSerializer(ModelSerializer):
+    artist = ArtistSerializer()
+    album = AlbumSerializer()
+
+    class Meta:
+        model = Track
+        fields = ("id", "title", "length", "artist", "album")
+
+    def create(self, validated_data):
+        artist_data = validated_data.pop("artist")
+        artist = Artist.objects.create(**artist_data)
+
+        album_data = validated_data.pop("album")
+        album = Album.objects.create(**album_data)
+
+        validated_data["artist"] = artist
+        validated_data["album"] = album
+
+        track = Track.objects.create(**validated_data)
+
+        return track
+
+    def update(self, instance, validated_data):
+        artist_data = validated_data.pop("artist")
+        if artist_data:
+            artist = instance.artist
+            artist.name = artist_data.get("name", artist.name)
+            artist.save()
+
+        album_data = validated_data.pop("album")
```

Comment:
Optimize update() method
In the TrackSerializer's update() method, you could optimize the code by using get() method of the dict object to get the values of artist_data and album_data, and also only update the artist and album fields if they exist in the incoming data.

artist_data = validated_data.get("artist")
        if artist_data:
            artist = instance.artist
            artist.name = artist_data.get("name", artist.name)
            artist.save()

        album_data = validated_data.get("album")
        if album_data:
            album = instance.album
            album.title = album_data.get("title", album.title)
            album.save()

        return super().update(instance, validated_data)